src/wp-includes/media.php

```diff
@@ -793,6 +810,117 @@ function wp_get_attachment_image($attachment_id, $size = 'thumbnail', $icon = fa
 }
 
 /**
+ * Return an array of urls and pixel widths representing sizes of an image for
+ * the purpose of populating a source set when creating responsive image markup.
+ *
+ * @since 4.4.0
+ *
+ * @param  int    $id   Image attachment ID.
+ * @param  string $size Optional. Name of image size. Default value: 'medium'.
+ * @return array|bool  An array of image urls and widths or false on failure.
```

Comment:
Looking into how do do this one correctly, and I'm not seeing any other functions that return a multi-dimensional array where the key is an integer, so I'm not sure what's best to do here. 

To be clear, the returned array looks like this:
```
  150 => 'http://test.dev/wp-content/uploads/2015/09/testimage-150x150.jpg 150w'
  300 => 'http://test.dev/wp-content/uploads/2015/09/testimage-300x300.jpg 300w'
  600 => 'http://test.dev/wp-content/uploads/2015/09/testimage.jpg 600w'
```

Do we rewrite the function to output a different array format and provide a compatibility shim in our plugin, or do we reach out to Drew about how to best format this?

For example, we could reformat these arrays to read like this:

```
[0] {
  'url'             => 'http://test.dev/wp-content/uploads/2015/09/testimage-150x150.jpg'
  'width'        => 150
  'descriptor => 'w'
},
[1] {
  'url'             => 'http://test.dev/wp-content/uploads/2015/09/testimage-300x300.jpg'
  'width'        => 300
  'descriptor => 'w'
},
[2] {
  'url'             => 'http://test.dev/wp-content/uploads/2015/09/testimage.jpg'
  'width'        => 600
  'descriptor => 'w'
}

```

I probably prefer reformatting the array, since that will be cleaner going forward, and a compat shim should be fairly easy to write.